From 5aa9f59275ef1447ffe0f5ebe080e11cc4bc740d Mon Sep 17 00:00:00 2001
From: Thomas Stewart <stewart@schrodinger.com>
Date: Wed, 7 Jul 2021 15:28:02 -0400
Subject: [PATCH] PYMOL-3694 Thumbnail using SceneUpdate - Fixes double
 recalling of scenes - Fixes crash when using middle mouse   switching between
 scenes

---
 layer3/MovieScene.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/layer3/MovieScene.cpp b/layer3/MovieScene.cpp
index 3f759891..2591abbd 100644
--- a/layer3/MovieScene.cpp
+++ b/layer3/MovieScene.cpp
@@ -233,7 +233,7 @@ pymol::Result<> MovieSceneStore(PyMOLGlobals * G, const char * name,
   int thumbnail_height = 100;
   scene.thumbnail = pymol::Image(thumbnail_width, thumbnail_height);
   png_outbuf_t png_buf;
-  ExecutiveDrawNow(G);
+  SceneUpdate(G, false); // Make sure scene is on screen
   SceneDeferImage(G, scene.thumbnail.getWidth(), scene.thumbnail.getHeight(),
       nullptr, 0, -1, 0, 1, &png_buf);
   scene.thumbnail.setVecData(png_buf);
-- 
2.32.0

